content/terraform/v1.12.x/docs/language/block/provider.mdx

```diff
@@ -0,0 +1,419 @@
+
+```hcl
+provider "google" {
+  project = "acme-app"
+  region  = "us-central1"
+}
+```
+
+</CodeBlockConfig>
+
+### Multiple provider configurations
+
+In the following example, two configurations for the AWS provider support different regions. The default configuration targets the east coast, and the aliased configuration targets the west coast:
+
+<CodeBlockConfig filename="main.tf">
+
+```hcl
+provider "aws" {
+  region = "us-east-1"
+}
+
+provider "aws" {
+  alias  = "west"
+  region = "us-west-2"
+}
+```
+
+</CodeBlockConfig>
+
+Resources that begin with `aws_` use the default `aws` provider configuration unless you supply the [`provider` argument](/terraform/language/block/resource#provider).
+
+### Using an alternate provider configuration
+
+You can make a resource, data source, or module use an alternate provider configuration by setting the `provider` meta-argument to a `<PROVIDER_NAME>.<ALIAS>` reference.
+
+In the following example, the `aws_instance.foo` resource uses the `aws.west` provider configuration, and `aws_instance.bar` uses the default `aws` provider configuration:
+
+<CodeBlockConfig highlight="6,11">
+
+```hcl
+provider "aws" {
+  region = "us-east-1"
+}
+
+provider "aws" {
+  alias  = "west"
+  region = "us-west-2"
+}
+
+resource "aws_instance" "foo" {
+  provider = aws.west
+  # …
+}
+
+resource "aws_instance" "bar" {
+   # …
+}
+```
+
+</CodeBlockConfig>
+
+### Provider without a default configuration
+
+If there are multiple aliases for a provider in your configuration, the `provider` block without an `alias` argument is the default configuration for that provider. If every `provider` block in your configuration uses an alias, Terraform creates an implied empty default configuration for that provider. Any resource that does not specify a `provider` meta-argument uses the empty default configuration.
+
+In the following example, both AWS provider configurations use aliases, but the `aws_s3_bucket` does not specify a `provider` argument:
+
+```hcl
+provider "aws" {
+  alias  = "east"
+  region = "us-east-1"
+}
+
+provider "aws" {
+  alias  = "west"
+  region = "us-west-2"
+}
+
+resource "aws_s3_bucket" "default_provider" {
+  bucket = "uses-implied-empty-config"
+}
+```
+
+Terraform uses the implied empty default configuration for the `aws_s3_bucket` resource because the resource doesn't specify a `provider` argument. If a provider requires specific arguments, Terraform returns an error when resources try to use the default configuration because the default is not properly configured.
+
+### Pass provider configurations to a child module
+
+We recommend defining providers in root modules, and passing those provider configurations from parent modules to child modules to ensure your modules all use the same provider configurations. You can pass provider configurations from a root module to child modules using the [`providers` argument](/terraform/language/meta-arguments/module-providers).
```

Comment:
This implies that we recommend explicitly passing providers to child modules rather than relying on implicit passing...I don't think we intend to do that, but I could be wrong...eg the main [use modules tutorial](https://developer.hashicorp.com/terraform/tutorials/modules/module-use#review-configuration) passes the provider implicitly.

Also, I'm not sure how much is recommendation vs. requirement, eg our [current docs say](https://developer.hashicorp.com/terraform/language/modules/develop/providers): `Provider configurations can be defined only in a root Terraform module.` - I don't think that's technically correct, but I think our intent is "Don't do this" rather than a soft recommendation.

Reply:
Also also, assuming implicitly passing providers is ok, and the default behavior is fine, then we should introduce an example of that first, before explicit example.